app/components/EditProfileForm.tsx

```diff
@@ -1,17 +1,59 @@
 'use client'
 import { FormEvent } from 'react';
+import { useRouter } from 'next/navigation';
 
-export default function EditProfileForm() {
+export default function EditProfileForm({userData}) {
 
-  /**
-   * Submits new post data
-   * 
-   * @param e - the form submit event
-  */
-  const onSubmitHandler = (e: FormEvent<HTMLFormElement>): void => {
+  const router = useRouter();
+
+  const firstName = userData['data'][0]['firstName'][0]
+  const lastName = userData['data'][0]['lastName'][0]
+  const username = userData['data'][0]['username'][0]
+  const gender = userData['data'][0]['gender'][0]
+  const bio = userData['data'][0]['bio'][0]
+
+  const userInfo = {
+    'firstName' : firstName,
+    'lastName' : lastName,
+    'username' : username,
+    'gender' : gender,
+    'bio' : bio
+  };
+
+  const onSubmitHandler = async (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     const formData: Object = Object.fromEntries((new FormData(e.currentTarget)));
-    console.log(JSON.stringify(formData));
+    
+    let attributes = {};
+
+    for (const [key, value] of Object.entries(formData)) {
+      // TODO: Add support for profile picture
+      if (key === "profilePicture") {
+        continue;
+      }
+      // If the value is the same, don't add to the request
+      if (userInfo[key] === value) {
+        continue;
+      } else {
+        attributes[key] = value;
+      }
+    }
+
+    const payload = {
+      "method" : "PUT",
+      "attributes" : attributes
+    }
+
+    const res = await fetch(`http://localhost:3000/api/users/${username}`, {
+          method: 'PUT',
+          headers: {'Content-Type': 'application/json'},
+          body: JSON.stringify(payload)
+        });
+
+    const data = await res.json();
+    console.log(data)
+
+    router.refresh();
```

Comment:
On success status code, instead of refresh lets redirect the user back to their dashboard.